ShareX.HelpersLib/Helpers/Helpers.cs

```diff
@@ -979,21 +979,68 @@ public static void CreateDirectoryFromFilePath(string filePath)
             }
         }
 
-        public static bool IsValidFilePath(string path)
+        public static bool IsValidFilePath(string path, bool directoryMustExist = false, bool showExceptions = false)
         {
             FileInfo fi = null;
 
             try
             {
                 fi = new FileInfo(path);
+
+                if (directoryMustExist && !Directory.Exists(Directory.GetParent(Path.GetFullPath(path)).FullName))
+                {
+                    return false;
+                }
+            }
+            catch (Exception e)
+            {
+                if (e is ArgumentException
+                    || e is PathTooLongException
+                    || e is NotSupportedException)
+                {
+                    MessageBox.Show($"Provided path: {path} is invalid!", "ShareX - Path Validation Error " + Resources.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
+                }
+
             }
-            catch (ArgumentException) { }
-            catch (PathTooLongException) { }
-            catch (NotSupportedException) { }
```

Comment:
Is there any reason for removing these error catchers?

Reply:
@kadamiec ?

Reply:
They are not removed, just handled correctly together with a message few lines above, as there they would be silent.
Sorry for late response.